src/s3.rs

```diff
@@ -54,18 +58,193 @@ pub fn get_client(shared_config: &aws_config::Config) -> Result<Client> {
     Ok(Client::from_conf(builder.build()))
 }
 
+/// Perform a bucket listing, returning a stream of results.
+///
+/// # Example
+///
+/// ```no_run
+/// use aws_config;
+/// use cobalt_aws::s3::{get_client, list_objects};
+/// use futures::TryStreamExt;
+///
+/// # tokio_test::block_on(async {
+/// let shared_config = aws_config::load_from_env().await;
+/// let client = get_client(&shared_config).unwrap();
+/// let mut objects = list_objects(&client, "my-bucket", Some("prefix".into()));
+/// while let Some(item) = objects.try_next().await.unwrap() {
+///     println!("{:?}", item);
+/// }
+/// # })
+/// ```
+///
+///  # Implementation details
+///
+/// This function uses the [ListObjectsV2](https://docs.aws.amazon.com/AmazonS3/latest/API/API_ListObjectsV2.html)
+/// API and performs pagination to ensure all objects are returned.
+pub fn list_objects(
+    client: &Client,
```

Comment:
I don't recall where we landed (if anywhere) on a broad design decision about using top-level functions with dependency-injected `Client` objects, versus using methods on a `Client` wrapper object.

I would personally favour this being a method on some object rather than a stand-alone top-level function, but I *even more* favour trying things out and learning from experience, so a simple top-level function seems like a good place to start :-)